k8s/base/tekton/resources/task-generate-cyclonedx.yaml

```diff
@@ -22,47 +22,114 @@ metadata:
   name: sbomer-generate-cyclonedx
 spec:
   params:
-    - name: jdk-version
-      type: string
-    - name: mvn-version
-      type: string
-    - name: workdir
-      type: string
-    - name: additional-cyclonedx-args
-      type: string
-      default: "--batch-mode --no-transfer-progress --quiet"
-    - name: cyclonedx-version
-      type: string
-      default: "2.7.5"
     - name: build-id
       type: string
+      description: "The PNC build ID"
+    # TODO: Currently that JSON is not really easy to work with in a Bash script, but this will change once we move to a proper CLI.
+    - name: config
+      type: string
+      description: |
+        Additional parameters that instrument the CycloneDX plugin in th JSON format:
+
+        Currently only following keys are supported:
+
+          * version:            Version of the Maven CycloneDX plugin, default: 2.7.5
+          * additional-args:    Additional arguments passed to the Maven CycloneDX plugin invocation
+
+        Example (and default):
+
+        {"version": "2.7.5", "additional-args": "--batch-mode --no-transfer-progress --quiet"}
+      default: |
+        {"version": "2.7.5", "additional-args": "--batch-mode --no-transfer-progress --quiet"}
+
+
+    # TODO: Content available below is not supported in the Tekton version we use.
+    # For more information please refer to this TEP: https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md
+
+    # - name: config
+    #   type: object
+    #   description: |
+    #     Additional parameters that instrument the CycloneDX plugin, available options:
+
+    #     * version:            Version of the Maven CycloneDX plugin, default: 2.7.5
+    #     * additional-args:    Additional arguments passed to the Maven CycloneDX plugin invocation
+    #   properties:
+    #     version:
+    #      type: string
+    #     additional-args:
+    #      type: string
+    #   default:
+    #     version: "2.7.5"
+    #     additional-args: "--batch-mode --no-transfer-progress --quiet"
   steps:
-    - name: generate-cyclonedx-sbom
+    # TODO: This is duplicated across every generation Task
+    # TODO: Move implementation of this step into Java CLI
+    - name: checkout
       image: sbomer-generator
       imagePullPolicy: IfNotPresent
       script: |
         #!/usr/bin/env bash
 
+        set -e
+
+        # Fetch build information
+        curl https://orch.psi.redhat.com/pnc-rest/v2/builds/$(params.build-id) | jq > $(workspaces.data.path)/build.json
```

Comment:
should we expose the orch URL inside the config?

Reply:
This is very temporary until we convert this into CLI, then we will be running a Quarkus CLI there, with full support for configuration.

Reply:
> Probably the orch URL could be another one to put already?

I don't see this URL being something that users should control. We will be, since we will be pointing to the proxy. See the point about CLI.

Reply:
right those should be user exposed configs. ok all good then!